File: Data Structures & Algorithms/trapping-rain-water/submission-4.py
class Solution:
    def trap(self, height: List[int]) -> int:
        n = len(height)

        res = 0
        for i in range(1, n-1):
            leftMax = height[i]
            for j in range(i-1, -1, -1):
                leftMax = max(leftMax, height[j])

            rightMax = height[i]
            for j in range(i+1, n):
                rightMax = max(rightMax, height[j])

            res += min(leftMax, rightMax) - height[i]

        return res        

File: Data Structures & Algorithms/max-water-container/submission-4.java
class Solution {
    public int maxArea(int[] heights) {
        int n = heights.length;

        int res = 0;
        int l=0, r = n-1;
        while(l < r){
            int min = Math.min(heights[l], heights[r]);
            res = Math.max(res, (r-l) * min);

            if(heights[l] < heights[r]){
                l++;
            } else {
                r--;
            }
        }

        return res;
    }
}
